### PR TITLE
Task/query point in polygon/337

### PR DESCRIPTION
**Pull request by andrea-williams**
Modified station_csv_to_json.py to incorporate ecodivision name & core fire season for each weather station

### CODE REVIEW

File: scripts/station_csv_to_json.py [hunk 1]
```diff
@@ -33,12 +48,19 @@
                     row[STATION_CODE], row[STATION_NAME]))
                 continue
             STATION_COUNT = STATION_COUNT + 1
+            ecodivision_name = fetch_ecodivision_name(row[STATION_LATITUDE], row[STATION_LONGITUDE], ECODIVISIONS)
+            if ecodivision_name is not None:
+                core_season = CORE_SEASONS[ecodivision_name]['core_season']
+            else:
+                core_season = None #pylint: disable=invalid-name
```
Comment (Kyubinhan) [inline scripts/station_csv_to_json.py:55]: It's kinda funny that we have to disable pylint for this. Is there some kind of conventions when using uppercase for variables in Python?

Comment (Sybrand) [reply scripts/station_csv_to_json.py:55]: I was trying to conform to UPPERCASE for constants - but I'm ready to through that out completely. Python doesn't have constants - so I'm ready to just change that in our linting settings and forget about uppercase variables.

File: pyproject.toml [hunk 2]
```diff
@@ -18,6 +18,7 @@ psycopg2 = "^2.8.5"
 postgres = "^3.0.0"
 python-decouple = "^3.3"
 uvicorn = "^0.11.5"
+geopandas = "^0.7.0"
```
Comment (Sybrand) [inline pyproject.toml:21]: Can you rather add this to dev-dependencies? The API doesn't need this, so I'd rather we just require it on our dev machines.

File: scripts/station_csv_to_json.py [hunk 3]
```diff
@@ -5,8 +5,19 @@
...
+def fetch_ecodivision_name(lat: str, long: str, ecodivisions: geopandas.GeoDataFrame):
+    """ Returns the ecodivision name for a given lat/long coordinate """
+    station_coord = Point(float(long), float(lat))
+    for index, row in ecodivisions.iterrows(): #pylint: disable=redefined-outer-name, unused-variable
+        geom = row['geometry']
+        if station_coord.within(geom):
+            return row['CDVSNNM']
+    return None
+
+with open('app/csv/Station_BC.csv', 'r') as csvfile:
```
Comment (Sybrand) [inline scripts/station_csv_to_json.py:20]: Can we keep the csv folder one level down please?  So just have it csv/Station_BC.csv?

File: scripts/station_csv_to_json.py [hunk 4]
```diff
@@ -18,6 +29,10 @@
 
     WEATHER_STATIONS = []
 
+    ECODIVISIONS = geopandas.read_file('app/data/ERC_ECODIV_polygon/ERC_ECODIV_polygon.shp')
```
Comment (Sybrand) [inline scripts/station_csv_to_json.py:32]: Can we move the shapefiles to a different folder please? It doesn't have to deploy as part of the app - unless you need it this way for local development in docker?

File: scripts/station_csv_to_json.py [hunk 5]
```diff
@@ -18,6 +29,10 @@
...
+    with open('app/data/ecodivisions_core_seasons.json') as file_handle:
```
Comment (Sybrand) [inline scripts/station_csv_to_json.py:33]: Can we keep this out of the app directory also? I don't see it referenced anywhere that the API needs it right now. (Just to keep what we deploy to the server to a minimum)

**Pull request merged**